libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -429,11 +429,13 @@ - (void)logoutUser:(SFUserAccount *)user
         return;
     }
 
-    if (user.isUserLoggingOut) {
-        [self log:SFLogLevelInfo msg:@"logoutUser: user is already in the process of logout."];
-        return;
+    @synchronized (user) {
```

Comment:
> I would prefer if we used a single write-multi read pattern around the state

We could do that, encapsulated in the user account object. But it will require additional "blessed" state transition checks. E.g. you can go from LoggedIn to LoggingOut, but not LoggingOut to LoggingOut, etc.

I can take a look at doing that.  Shouldn't be a big change.  Stay tuned.